core/src/main/java/hudson/Util.java

```diff
@@ -251,18 +257,8 @@ public static void deleteFile(@Nonnull File f) throws IOException {
 
             if(!f.delete() && f.exists()) {
                 // trouble-shooting.
-                try {
-                    Class.forName("java.nio.file.Files").getMethod("delete", Class.forName("java.nio.file.Path")).invoke(null, File.class.getMethod("toPath").invoke(f));
-                } catch (InvocationTargetException x) {
-                    Throwable x2 = x.getCause();
-                    if (x2 instanceof IOException) {
-                        // may have a specific exception message
-                        throw (IOException) x2;
-                    }
-                    // else suppress
-                } catch (Throwable x) {
-                    // linkage errors, etc.; suppress
-                }
+                Files.delete(f.toPath());
+
                 // see http://www.nabble.com/Sometimes-can%27t-delete-files-from-hudson.scm.SubversionSCM%24CheckOutTask.invoke%28%29-tt17333292.html
                 // I suspect other processes putting files in this directory
                 File[] files = f.listFiles();
```

Comment:
:bug: this code is now in effect unreachable - if the file can not be deleted then an exception would be thrown.  If it can be deleted then this code is a no-op